packages/aws-cdk/lib/settings.ts

```diff
@@ -62,6 +68,40 @@ export class Configuration {
  * A single set of settings
  */
 export class Settings {
+  /**
+   * Parse Settings out of CLI arguments.
+   * @param argv the received CLI arguments.
+   * @returns a new Settings object.
+   */
+  public static fromCommandLineArguments(argv: yargs.Arguments): Settings {
+    const context: any = {};
+
+    // Turn list of KEY=VALUE strings into an object
+    for (const assignment of (argv.context || [])) {
+      const parts = assignment.split('=', 2);
+      if (parts.length === 2) {
+        debug('CLI argument context: %s=%s', parts[0], parts[1]);
+        if (parts[0].match(/^aws:.+/)) {
+          throw new Error(`User-provided context cannot use keys prefixed with 'aws:', but ${parts[0]} was provided.`);
+        }
+        context[parts[0]] = parts[1];
+      } else {
+        warning('Context argument is not an assignment (key=value): %s', assignment);
+      }
+    }
+
+    return new Settings({
+      app: argv.app,
+      browser: argv.browser,
+      context,
+      language: argv.language,
+      plugin: argv.plugin,
+      toolkitStackName: argv.toolkitStackName,
```

Comment:
Looks to me you've just moved the code here. What am I missing?

I assumed you would make it so all yargs arguments could be loaded from context. Is that in the cards at all?

Reply:
Right now - just moving around so you don't have to "bring your own" parser anymore. Before I go and make this "more prevalent" I'd want to formalize the schema of `cdk.json`...